### PR TITLE
FAQGen Megaservice

### PR DESCRIPTION
**Pull request by yogeshmpandey**
## Description

This PR introduces FAQGen Megaservice in GenAIExamples. Dependent on PR [318
](https://github.com/opea-project/GenAIComps/pull/318) in GenAIComps

## Issues

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Test files to test on Xeon and Gaudi nodes

### CODE REVIEW

**Review by amberjain1 — APPROVED** [review]

**Comment by chensuyue** [comment]
CI for this image only able to be test after https://github.com/opea-project/GenAIComps/pull/318 merged.

**Comment by yogeshmpandey** [comment]
> CI for this example only able to be test after [opea-project/GenAIComps#318](https://github.com/opea-project/GenAIComps/pull/318) merged.

@chensuyue I have the[ tests updated](https://github.com/opea-project/GenAIExamples/pull/425/files#diff-c74dbce427a0081a6efad37aca20663a0f9879260f874010a2a9e9879cd9b810) to pull the Comps from the PR itself, the tests run fine locally on Xeon and Gaudi. Not sure why they are failing here.

File: FaqGen/tests/test_faqgen_on_gaudi.sh [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ip_address=$(hostname -I | awk '{print $1}')
+
+function build_docker_images() {
+    cd $WORKPATH
+
+    git clone https://github.com/opea-project/GenAIComps.git
+    cd GenAIComps
+    docker build --no-cache -t opea/llm-faqgen-tgi:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/llms/faq-generation/tgi/Dockerfile .
+
+    docker pull ghcr.io/huggingface/tgi-gaudi:1.2.1
```
Comment (chensuyue) [inline FaqGen/tests/test_faqgen_on_gaudi.sh:18]: The habana driver version of the test machine is v1.16.1, the tgi-gaudi should be ghcr.io/huggingface/tgi-gaudi:2.0.1, I think that may be the issue why test failed.

Comment (chensuyue) [reply FaqGen/tests/test_faqgen_on_gaudi.sh:18]: Artifact download URL: https://github.com/opea-project/GenAIExamples/actions/runs/10057922170/artifacts/1730592345

Comment (yogeshmpandey) [reply FaqGen/tests/test_faqgen_on_gaudi.sh:18]: Hi @chensuyue , thanks updated the tgi-gaudi  Version
All tests are passed now

**Review by WenjiaoYue — COMMENTED** [review]
The UI code (React, Svelte) is almost just renamed and copied from the content of docSum. So, what is the purpose of FAQGen? What improvements have been made, and are these improvements necessary to be provided to users as a complete repository? If the modifications are minor, can improvements be made in docSum or explained in docSum?

**Comment by yogeshmpandey** [comment]
> The UI code (React, Svelte) is almost just renamed and copied from the content of docSum. So, what is the purpose of FAQGen? What improvements have been made, and are these improvements necessary to be provided to users as a complete repository? If the modifications are minor, can improvements be made in docSum or explained in docSum?

@WenjiaoYue , Since FAQGen needs special prompts to generate FAQs from the data, it is currently implemented as a different example. There is an ongoing [discussion](https://github.com/opea-project/GenAIComps/pull/318#issuecomment-2242222388) on consolidating the LLMs in upcoming releases.  Later, we can update docsum to do summarization, FaqGen, meeting summary and other data insights from the same example.

**Review by ftian1 — APPROVED** [review]

**Pull request merged**